EIPS/eip-7378.md

```diff
@@ -0,0 +1,79 @@
+---
+eip: 7378
+title: Add time-weighted averaging to the base fee
+description: Using geometric weights to average past block sizes into consideration
+author: Guy Goren (@guy-goren) <guy.nahsholim@gmail.com>
+discussions-to: https://ethereum-magicians.org/t/add-time-weighted-averaging-to-the-base-fee-mechanism/15142
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-07-22
+---
+
+## Abstract
+
+This EIP proposes a new formula for the base fee, derived from [EIP-1559](./eip-1559.md). The existing base fee formula,
+
+$$b[i+1]\triangleq  b[i] \cdot \left( 1+\frac{1}{8} \cdot \frac{s[i]-s^* }{s^* }\right)$$
+
+only considers the last block size $s[i]$. This mechanism incentivizes proposers to collude with users to manipulate the base fee.
+
+We propose to consider the history of block sizes via a weighted average with a geometric sequence as the weights. In particular, we suggest the following formula:
```

Comment:
The wording "We propose to consider the history of block sizes via a weighted average with a geometric sequence as the weights." feels clunky. Additionally, the thing you're describing actually has a name: "exponential moving average". How about:

```suggestion
We propose that previous block sizes be considered by replacing the previous block size with an exponential moving average. In particular, we suggest the following formula:
```

The smoothing factor is also typically represented by $\alpha$